tests/ttnn/sweep_tests/shard_utility.py

```diff
@@ -0,0 +1,122 @@
+    width,
+    layout,
+    input_shard_orientation,
+    input_num_cores_x,
+    input_num_cores_y,
+    input_shard_strategy,
+    interleaved_to_sharded,
+    sharded_to_interleaved,
+    reshard,
+    device,
+    output_shard_orientation=None,
+    output_num_cores_x=None,
+    output_num_cores_y=None,
+    output_shard_strategy=None,
+):
+    input_num_cores_height = 1
+    input_num_cores_width = 1
+    if input_shard_strategy == ttnn.ShardStrategy.HEIGHT:
+        input_num_cores_height = input_num_cores_x * input_num_cores_y
+    elif input_shard_strategy == ttnn.ShardStrategy.WIDTH:
+        input_num_cores_width = input_num_cores_x * input_num_cores_y
+    elif input_shard_strategy == ttnn.ShardStrategy.BLOCK:
+        if input_shard_orientation == ttnn.ShardOrientation.ROW_MAJOR:
+            input_num_cores_height = input_num_cores_y
+            input_num_cores_width = input_num_cores_x
+        else:
+            input_num_cores_height = input_num_cores_x
+            input_num_cores_width = input_num_cores_y
+
+    input_core_grid = ttnn.CoreGrid(y=input_num_cores_y, x=input_num_cores_x)
+    input_args = dict(
+        core_grid=input_core_grid,
+        strategy=input_shard_strategy,
+        orientation=input_shard_orientation,
+    )
+    num_cores_height = input_num_cores_height
+    num_cores_width = input_num_cores_width
+
+    if reshard:
+        output_num_cores_height = 1
+        output_num_cores_width = 1
+        if output_shard_strategy == ttnn.ShardStrategy.HEIGHT:
+            output_num_cores_height = output_num_cores_x * output_num_cores_y
+        elif output_shard_strategy == ttnn.ShardStrategy.WIDTH:
+            output_num_cores_width = output_num_cores_x * output_num_cores_y
+        elif output_shard_strategy == ttnn.ShardStrategy.BLOCK:
+            if output_shard_orientation == ttnn.ShardOrientation.ROW_MAJOR:
+                output_num_cores_height = output_num_cores_y
+                output_num_cores_width = output_num_cores_x
+            else:
+                output_num_cores_height = output_num_cores_x
+                output_num_cores_width = output_num_cores_y
+
+        num_cores_height = max(input_num_cores_height, output_num_cores_height)
+        num_cores_width = max(input_num_cores_width, output_num_cores_width)
+        output_core_grid = ttnn.CoreGrid(y=output_num_cores_y, x=output_num_cores_x)
+        output_args = dict(
+            core_grid=output_core_grid,
+            strategy=output_shard_strategy,
+            orientation=output_shard_orientation,
+        )
+
+    if layout == ttnn.TILE_LAYOUT:
+        height = int(roundTo32(height))
+        width = int(roundTo32(width))
+
+    height = height * num_cores_height
+    width = width * num_cores_width
+    tensor_shape = [1, 1, int(height), int(width)]
+
+    torch_input_tensor = torch.randn(tensor_shape, dtype=torch.bfloat16)
+    input_shard_memory_config = ttnn.create_sharded_memory_config(tensor_shape, **input_args)
```

Comment:
Can we not call this function? It looks janky and we should reduce dependencies and just directly specify what we want to test so it's clear what we are testing

Reply:
are you referring to the `create_sharded_memory_config` function?